platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -27,12 +27,37 @@ @implementation NSExpression (MGLAdditions)
         return { std::string([(NSString *)value UTF8String]) };
     } else if ([value isKindOfClass:NSNumber.class]) {
         NSNumber *number = (NSNumber *)value;
-        if((strcmp([number objCType], @encode(int))) == 0) {
+        // handle all types defined in NSNumber designated initializers
+        if (strcmp([number objCType], @encode(char)) == 0) {
+            return { number.charValue };
+        } else if (strcmp([number objCType], @encode(unsigned char)) == 0) {
+            return { number.unsignedCharValue };
+        } else if (strcmp([number objCType], @encode(short)) == 0) {
+            return { number.shortValue };
+        } else if (strcmp([number objCType], @encode(unsigned short)) == 0) {
+            return { number.unsignedShortValue };
+        } else if (strcmp([number objCType], @encode(int)) == 0) {
             return { number.intValue };
-        } else if ((strcmp([number objCType], @encode(double))) == 0) {
+        } else if (strcmp([number objCType], @encode(unsigned int)) == 0) {
+            return { number.unsignedIntValue };
+        } else if (strcmp([number objCType], @encode(long)) == 0) {
+            return { number.longValue };
+        } else if (strcmp([number objCType], @encode(unsigned long)) == 0) {
+            return { number.unsignedLongValue };
+        } else if (strcmp([number objCType], @encode(long long)) == 0) {
+            return { number.longLongValue };
+        } else if (strcmp([number objCType], @encode(unsigned long long)) == 0) {
+            return { number.unsignedLongLongValue };
+        } else if (strcmp([number objCType], @encode(float)) == 0) {
+            return { number.floatValue };
+        } else if (strcmp([number objCType], @encode(double)) == 0) {
             return { number.doubleValue };
-        } else if ((strcmp([number objCType], @encode(char))) == 0) {
+        } else if (strcmp([number objCType], @encode(bool)) == 0) {
```

Comment:
BOOL is another type that is represented differently between 32- and 64-bit systems. 7e208c46cebe5b49d4b7ead9826bbe4229ebabd3 broke Boolean conversions on 32-bit systems while fixing it for 64-bit systems; this change does the opposite, breaking 64-bit but fixing 32-bit. We need both cases, ideally predicated (hah!) on the architecture via conditional compilation.